src/app/shared/pagination/pagination.component.ts

```diff
@@ -147,209 +151,209 @@ export class PaginationComponent implements OnDestroy, OnInit {
+        pageId: this.id,
+        page: this.currentPage,
+        pageSize: this.pageSize,
+        sortDirection: this.sortDirection,
+        sortField: this.sortField
+      })
+    });
+  }
 
-    /**
-     * Validate query params
-     *
-     * @param page
-     *    The page number to validate
-     * @param pageSize
-     *    The page size to validate
-     */
-    private validateParams(page: any, pageSize: any, sortDirection: any, sortField: any) {
-        let filteredPageSize = this.pageSizeOptions.find(x => x == pageSize);
-        if (!isNumeric(page) || !filteredPageSize) {
-          let filteredPage = isNumeric(page) ? page : this.currentPage;
-          filteredPageSize = (filteredPageSize) ? filteredPageSize : this.pageSize;
-          this.router.navigate([], {
-              queryParams: {
-                pageId: this.id,
-                page: filteredPage,
-                pageSize: filteredPageSize,
-                sortDirection: sortDirection,
-                sortField: sortField
-              }
-            }
-          );
-        } else {
-            // (+) converts string to a number
-            this.currentPage = +page;
-            this.pageSize = +pageSize;
-            this.sortDirection = +sortDirection;
-            this.sortField = sortField;
-            this.pageChange.emit(this.currentPage);
-            this.pageSizeChange.emit(this.pageSize);
-            this.sortDirectionChange.emit(this.sortDirection);
-            this.sortFieldChange.emit(this.sortField);
-        }
+  /**
+   * Method to set pagination details of the current viewed page.
+   */
+  private setShowingDetail() {
+    let firstItem;
+    let lastItem;
+    const lastPage = Math.round(this.collectionSize / this.pageSize);
+
+    firstItem = this.pageSize * (this.currentPage - 1) + 1;
+    if (this.currentPage !== lastPage) {
+      lastItem = this.pageSize * this.currentPage;
+    } else {
+      lastItem = this.collectionSize;
+    }
+    this.showingDetail = {
+      range: firstItem + ' - ' + lastItem,
+      total: this.collectionSize
     }
+  }
 
-    /**
-     * Ensure options passed contains the required properties.
-     *
-     * @param paginateOptions
-     *    The paginate options object.
-     */
-    private checkConfig(paginateOptions: any) {
-        let required = ['id', 'currentPage', 'pageSize', 'pageSizeOptions'];
-        let missing = required.filter(function (prop) {
-            return !(prop in paginateOptions);
-        });
-        if (0 < missing.length) {
-            throw new Error("Paginate: Argument is missing the following required properties: " + missing.join(', '));
+  /**
+   * Validate query params
+   *
+   * @param page
+   *    The page number to validate
+   * @param pageSize
+   *    The page size to validate
+   */
+  private validateParams(page: any, pageSize: any, sortDirection: any, sortField: any) {
+    let filteredPageSize = this.pageSizeOptions.find((x) => x === pageSize);
```

Comment:
Casting variables to identical type would probably be better. Would allow exact comparison. For now I will disable the lint rule for these cases. Beyond the scope of the upgrade.

You are right the pre-prender does revert to the first page. Rehydrate is just faster in navigating to the correct page. I will attempt to fix this quickly.